### DISCUSSION TITLE
Vercel builds failing but no reason given

### DISCUSSION THREAD

**Discussion by pearcircuitmike** [vercel/vercel#9152] [Help]
I have started encountering a new Vercel issue. My site is built (~38k routes, SSG'd) in ~10 mins. However, the build then hangs and fails (succeeds locally, just fails on Vercel). I get the following message:

> An unexpected error happened when running this build. We have been notified of the problem. If you have any questions, please contact Vercel Support https://vercel.com/help.

This has happened a few times. Not sure what the issue is. Don't think it's timing out. 

Someone else had almost the exact same issue a few months ago, and it was just solved after retrying a few days later - the cause and fix weren't shared by either the poster or support team, so I can't use that information to solve this problem. You can read that issue here: https://github.com/vercel/vercel/discussions/8195

You can see it hangs in the logs right after "build completed"

> Build Completed in /vercel/output [11m]
> Generated build outputs:
> - Static files: 39
> - Serverless Functions: 76949
> - Edge Functions: 0
> Serverless regions: Washington, D.C., USA
> Deployed outputs in 534s
> An unexpected error happened when running this build. We have been notified of the problem. If you have any questions, please contact Vercel Support https://vercel.com/help.

Because the error message is so cryptic, with nothing in the logs, I opened a support ticket. However, the response I got back was:

> Thank you for reaching out. The build error you are seeing appears to be from your project generating too many routes and reaching an upper limit on Vercel. Could you try reducing the number of routes in your project? Could you let us know which Next.js data fetching methods your project is using?
> 
> The best path forward is to reduce the number of paths returned by getStaticPaths() in your project. **Using "revalidate" for ISR and having a high number of paths is what is causing the error.** That function getStaticPaths() returns a list of paths that all become lambda paths. The large number of serverless functions are not real serverless functions, but prerenders that are attached to serverless functions for [Incremental Static Generation](https://nextjs.org/docs/basic-features/data-fetching/incremental-static-regeneration). To reduce the number of paths you could also try some paths on demand https://nextjs.org/docs/basic-features/data-fetching/get-static-paths#generating-paths-on-demand

I really don't know what to make of this as there is no upper limit to the number of routes reported anywhere that I could find in the limits documentation, and I'm not using ISR, just pure SSG (I'm not passing a revalidate param).

My use case is a site with about 38k details page and some comparison pages (almost all static).

**Comment by amyegan** [comment]
Thanks for reaching out @pearcircuitmike. Do you have a public repository or [minimal reproducible example](https://vercel.com/support/articles/creating-a-minimal-reproducible-example) that you can share for debugging? A link to a recent failing deployment would also be helpful to allow us to dig into the logs.
